### PR TITLE
Add a tutorial explianing how to check the exactness of a certificate

### PR DESCRIPTION
**Pull request by g-goessel**
I wanted to check if a certificate was correct but failed at finding a documentation. After many attempts I found out that I had to use the verify method on the public key in order to manually check the signature.

### CODE REVIEW

**Comment by g-goessel** [comment]
Thank you for your explanation.
I know this is not all we have to check but I wanted to do the steps manually (for a school project actually).

**Comment by reaperhulk** [comment]
If you'd like to add the example code under the `tbs_certificate_bytes` section we'd be willing to take that as a PR. It would need to be done with `doctest` rather than `pycon` as well since we want to run our code during doc generation to ensure we always have functional examples.

File: docs/x509/reference.rst [hunk 1]
```diff
@@ -415,6 +463,29 @@ X.509 Certificate Object
...
+        on the public key (this does not mean that the cert should be truster):
+
+        .. doctest::
+
+           >>> from cryptography.hazmat.primitives.serialization import load_pem_public_key
+           >>> from cryptography.exceptions import InvalidSignature
+           >>> from cryptography.hazmat.primitives.asymmetric import padding
+           >>> issuer_public_key = load_pem_public_key(pem_issuer_public_key, default_backend())
+           >>> cert_to_check = x509.load_pem_x509_certificate(pem_data_to_check, default_backend())
+           >>> try:
```
Comment (reaperhulk) [inline docs/x509/reference.rst]: For the doctest we can just call the verify call directly, no need to pu tit in a try block.

Comment (g-goessel) [reply docs/x509/reference.rst]: hum ok, will change it
sorry for the failed CI, I made a copy/paste error...

File: docs/x509/reference.rst [hunk 2]
```diff
@@ -415,6 +462,29 @@ X.509 Certificate Object
         certificate validation is a complex problem that involves much more
         than just signature checks.
 
+        An example on how to check if a certificate signature is correct with a method
+        on the public key (this does not mean that the cert should be trusted):
+
+        .. doctest::
+
+           >>> from cryptography.hazmat.primitives.serialization import load_pem_public_key
+           >>> from cryptography.exceptions import InvalidSignature
```
Comment (reaperhulk) [inline docs/x509/reference.rst]: This import is no longer required

File: docs/x509/reference.rst [hunk 3]
```diff
@@ -415,6 +462,29 @@ X.509 Certificate Object
...
+           >>> from cryptography.hazmat.primitives.asymmetric import padding
+           >>> issuer_public_key = load_pem_public_key(pem_issuer_public_key, default_backend())
+           >>> cert_to_check = x509.load_pem_x509_certificate(pem_data_to_check, default_backend())
+           >>> # verify() raises a
```
Comment (reaperhulk) [inline docs/x509/reference.rst]: This comment can be removed (or made into a complete sentence)

File: docs/x509/reference.rst [hunk 4]
```diff
@@ -415,6 +462,29 @@ X.509 Certificate Object
...
+           ...   cert_to_check.signature,
+           ...   cert_to_check.tbs_certificate_bytes,
+           ...   #Depends on the algorithm used to create the certificate
+           ...   padding.PKCS1v15(),
+           ...   cert_to_check.signature_hash_algorithm,
+           ... )
+
+           An
+           :class:`~cryptography.exceptions.InvalidSignature`
+           exception will be raised otherwise.
```
Comment (reaperhulk) [inline docs/x509/reference.rst]: "will be raised if the signature fails to verify"

**Comment by g-goessel** [comment]
Oops made a mistake while rebasing, please stash the commits together when merging :smile:

**Pull request merged**